lxd/identity/cache.go

```diff
@@ -0,0 +1,151 @@
+package identity
+
+import (
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/api"
+)
+
+// Cache represents an thread-safe in-memory cache of the identities in the database.
+type Cache struct {
+	entries map[string]map[string]CacheEntry
```

Comment:
could you add a comment explaining what this map of maps is keyed on please

Reply:
My understanding is that maps cannot shrink the size of memory they use, so we should have them store *CachedEntry so that the map just stores a pointer rather than the whole struct.

Reply:
https://100go.co/28-maps-memory-leaks/

Reply:
Will do.